### PR TITLE
First PR Draft concerning data storage.

### PR DESCRIPTION
**Pull request by christophezei**
-Added json file structure to save needed data.
-Added logic to save the coordinates in this file.

### CODE REVIEW

File: gazeplay-commons/src/main/java/net/gazeplay/commons/utils/stats/Stats.java [hunk 1]
```diff
@@ -97,6 +95,9 @@
 
     private Long currentRoundStartTime;
 
+    private ArrayList<String> coordinateData = new ArrayList<String>();
+    private JsonObject savedDataObj=new JsonObject();
```
Comment (Tohaker) [inline gazeplay-commons/src/main/java/net/gazeplay/commons/utils/stats/Stats.java]: Could you format this file in your IDE to add spaces around `=` signs?

File: gazeplay-commons/src/main/java/net/gazeplay/commons/utils/stats/Stats.java [hunk 2]
```diff
@@ -234,6 +236,9 @@ public void start() {
             recordMouseMovements = e -> {
                 final int getX = (int) e.getX();
                 final int getY = (int) e.getY();
+                final String coordX=Integer.toString(getX);
+                final String coordY=Integer.toString(getY);
+                saveCoordinates("{" + coordX + "," + coordY + "}");
```
Comment (Tohaker) [inline gazeplay-commons/src/main/java/net/gazeplay/commons/utils/stats/Stats.java]: You could replace the String declarations with the `String.format()` method to add in the X and Y coordinates.

File: gazeplay-commons/src/main/java/net/gazeplay/commons/utils/stats/Stats.java [hunk 3]
```diff
@@ -316,11 +321,13 @@ public SavedStatsInfo saveStats() throws IOException {
         final String gazeMetricsFilePrefix = Utils.now() + "-metrics";
         final String screenShotFilePrefix = Utils.now() + "-screenshot";
         final String colorBandsFilePrefix = Utils.now() + "-colorBands";
+        final String replayDataFilePrefix= Utils.now() + "-replayData";
 
         final File gazeMetricsFile = new File(todayDirectory, gazeMetricsFilePrefix + ".png");
         final File heatMapCsvFile = new File(todayDirectory, heatmapFilePrefix + ".csv");
         final File screenShotFile = new File(todayDirectory, screenShotFilePrefix + ".png");
         final File colorBandsFile = new File(todayDirectory, colorBandsFilePrefix + "png");
+        File replayDataFile = new File(todayDirectory, replayDataFilePrefix + ".json");
```
Comment (Tohaker) [inline gazeplay-commons/src/main/java/net/gazeplay/commons/utils/stats/Stats.java]: Please make this `final File`

File: gazeplay-commons/src/main/java/net/gazeplay/commons/utils/stats/Stats.java [hunk 4]
```diff
@@ -578,4 +588,27 @@ private void takeScreenShot() {
...
+        savedDataObj.addProperty("coordinates",data.toString());
+        return savedDataObj;
+    }
+
+    private ArrayList<String> saveCoordinates (String coordinates){
+        coordinateData.add(coordinates);
+        return coordinateData;
+    }
+
+    private int randNumb(int limit)
```
Comment (Tohaker) [inline gazeplay-commons/src/main/java/net/gazeplay/commons/utils/stats/Stats.java]: Could this be renamed to `randNumber` to make it clearer?

**Comment by Tohaker** [comment]
You have some SpotBugs issues, could you check these?

**Comment by kuhlkrein** [comment]
Hi @christophezei,  looks like you have some conflicts with develop

**Comment by kuhlkrein** [comment]
This will be done in another PR

**Pull request closed without merging**